src/cursor/cur_version.c

```diff
@@ -0,0 +1,464 @@
+        /*
+         * If the update is an aborted update, we want to skip to the next update immediately or get
+         * the ondisk value if the update is the last one in the update chain.
+         */
+        while (upd != NULL && upd->txnid == WT_TXN_ABORTED)
+            upd = upd->next;
+
+        if (upd == NULL)
+            F_SET(version_cursor, WT_VERSION_CUR_UPDATE_EXHAUSTED);
+        else {
+            /*
+             * If the update is a tombstone, we still want to record the stop information but we
+             * also need traverse to the next update to get the full value. If the tombstone was the
+             * last update in the update list, retrieve the ondisk value.
+             */
+            version_cursor->upd_stop_txnid = upd->txnid;
+            version_cursor->upd_durable_stop_ts = upd->durable_ts;
+            version_cursor->upd_stop_ts = upd->start_ts;
+            if (upd->type == WT_UPDATE_TOMBSTONE) {
+                upd = upd->next;
+
+                /* Make sure the next update is not an aborted update. */
+                while (upd != NULL && upd->txnid == WT_TXN_ABORTED)
+                    upd = upd->next;
+            }
+
+            if (upd == NULL)
+                F_SET(version_cursor, WT_VERSION_CUR_UPDATE_EXHAUSTED);
+            else {
+                if (upd->prepare_state == WT_PREPARE_INPROGRESS ||
+                  upd->prepare_state == WT_PREPARE_LOCKED)
+                    version_prepare_state = 1;
+                else
+                    version_prepare_state = 0;
+
+                /*
+                 * Set the version cursor's key, which contains all the record metadata for that
+                 * particular version of the update.
+                 */
+                __wt_cursor_set_key(cursor, upd->txnid, upd->start_ts, upd->durable_ts,
+                  version_cursor->upd_stop_txnid, version_cursor->upd_stop_ts,
+                  version_cursor->upd_durable_stop_ts, upd->type, version_prepare_state, upd->flags,
+                  WT_VERSION_UPDATE_CHAIN);
+
+                /*
+                 * Copy the update value into the version cursor as we don't know the value format.
+                 * If the update is a modify, reconstruct the value.
+                 */
+                if (upd->type != WT_UPDATE_MODIFY)
+                    __wt_upd_value_assign(cbt->upd_value, upd);
+                else
+                    WT_ERR(
+                      __wt_modify_reconstruct_from_upd_list(session, cbt, upd, cbt->upd_value));
+
+                cursor->value.data = cbt->upd_value->buf.data;
+                cursor->value.size = cbt->upd_value->buf.size;
+                F_SET(cursor, WT_CURSTD_VALUE_INT);
+
+                upd_found = true;
+                version_cursor->next_upd = upd->next;
+            }
+        }
+    }
+
+    if (!upd_found && !F_ISSET(version_cursor, WT_VERSION_CUR_ON_DISK_EXHAUSTED)) {
+        WT_TIME_WINDOW_INIT(&vpack->tw);
```

Comment:
No need for this.

Reply:
This is here to silence some compiler warnings about potential garbage value.